3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/PiiPseudonymizerEnrichment.scala

```diff
@@ -0,0 +1,342 @@
+      event.user_fingerprint = fn(event.user_fingerprint)
+    },
+    "domain_userid" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.domain_userid = fn(event.domain_userid)
+    },
+    "network_userid" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.network_userid = fn(event.network_userid)
+    },
+    "ip_organization" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.ip_organization = fn(event.ip_organization)
+    },
+    "ip_domain" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.ip_domain = fn(event.ip_domain)
+    },
+    "tr_orderid" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.tr_orderid = fn(event.tr_orderid)
+    },
+    "ti_orderid" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.ti_orderid = fn(event.ti_orderid)
+    },
+    "mkt_term" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.mkt_term = fn(event.mkt_term)
+    },
+    "mkt_content" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.mkt_content = fn(event.mkt_content)
+    },
+    "se_category" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.se_category = fn(event.se_category)
+    },
+    "se_action" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.se_action = fn(event.se_action)
+    },
+    "se_label" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.se_label = fn(event.se_label)
+    },
+    "se_property" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.se_property = fn(event.se_property)
+    },
+    "mkt_clickid" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.mkt_clickid = fn(event.mkt_clickid)
+    },
+    "refr_domain_userid" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.refr_domain_userid = fn(event.refr_domain_userid)
+    },
+    "domain_sessionid" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.domain_sessionid = fn(event.domain_sessionid)
+    }
+  )
+
+  val JsonMutators: Map[String, Mutator] = Map(
+    "contexts" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.contexts = fn(event.contexts)
+    },
+    "derived_contexts" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.derived_contexts = fn(event.derived_contexts)
+    },
+    "unstruct_event" -> { (event: EnrichedEvent, fn: String => String) =>
+      event.unstruct_event = fn(event.unstruct_event)
+    }
+  )
+}
+
+/**
+ * PiiField trait. This corresponds to a configuration top-level field (i.e. either a POJO or a JSON field) along with
+ * a function to apply that strategy to a TransformMap.
+ */
+sealed trait PiiField {
+  import PiiConstants.Mutator
+
+  /**
+   * Strategy for this field
+   *
+   * @return PiiStrategy a strategy to be applied to this field
+   */
+  def strategy: PiiStrategy
+
+  /**
+   * The POJO mutator for this field
```

Comment:
This is for pojo and json now, no?

Reply:
json is just another field (it contains a json string instead of a scalar) in the POJO, so at this level they are both POJO mutators the way I see it. Maybe the we should rename the Pojo mutator to "scalar mutator" to make that clear, What do you think?

Reply:
I kinda prefer scalar mutator but I don't mind either way.

Reply:
scalar mutator sounds more accurate to me. I am changing it.